### DISCUSSION TITLE
How to input the temperature field from FLOW3D into MOOSE to simulate grain evolution

### DISCUSSION THREAD

**Discussion by zsren** [idaholab/moose#22511] [Q&A Modules: Phase field] [answered]
Hi,

I am new to MOOSE. I am trying to simulate the grain growth of laser powder bed fusion by utilizing the phase field module in MOOSE. I plan to input the temperature field, which is calculated within a thermal-hydrodynamic simulation from another software FLOW3D, into MOOSE. Is there by any chance an interface to make this coupling possible? Or is there a specific data format, with which FLOW3D should be output, such that MOOSE can recognize? Some existing examples would be super helpful!

It would be greatly appreciated for your efforts in making any comments or suggestions!

ZR

**Comment by GiudGiud** [comment, marked as answer]
Hello

What are all the formats in which you can output data from FLOW3D?
What kind of data is it going to be? A xyz field with a time dependence?

Guillaume

**Reply by zsren** [reply]
Hi Guillaume,

Thanks so much for your prompt response. As far as I know, FLOW3D can export the X, Y, and Z coordinates with temperature values at a particular timestep in a CSV file.

ZR

**Reply by GiudGiud** [reply]
Sorry for the delay.

PiecewiseConstantFromCSV is the function you will want to use
